### ISSUE TITLE
YAML Multiline

### ISSUE THREAD

**Issue by iafelix** [evilmartians/lefthook#460]
### :zap: Summary

Would be nice to have YAML multiline strings for `run` option following the YAML specs:
[https://yaml-multiline.info/](https://yaml-multiline.info/)

### Value

It is hard to read long lines of commands for instance:

```yaml
commands:
  trivy:
    tags: security
    run: docker run --rm --volume "$(git rev-parse --show-toplevel)":/home/docker cdci trivy filesystem --quiet --ignore-unfixed --ignorefile .trivyignore --skip-files .env /home/docker
```

### Behavior and configuration changes

That long line could be written like so:

```yaml
commands:
  trivy:
    tags: security
    run: |
      docker run --rm --volume "$(git rev-parse --show-toplevel)":/home/docker cdci 
      trivy filesystem --quiet --ignore-unfixed --ignorefile .trivyignore --skip-files .env /home/docker
```

**Comment by technicalpickles** [comment]
What happens if you try to use use multi-line currently? Looking at `gopkg.in/yaml.v3 v3.0.1` in `go.mod`, it seems like it should be able to use them based on https://github.com/go-yaml/yaml/commit/513397e5a74045296989ac1a3650ae511bf96dab being part of v3.0.1

From a casual reading of your example, I'd expect it to treat separate lines as different commands, since that is how multiple lines in a shell would execute. Using a `\` to span commands multiple lines might work:

```yaml
commands:
  trivy:
    tags: security
    run: |
      docker run --rm --volume "$(git rev-parse --show-toplevel)":/home/docker cdci \
        trivy filesystem --quiet --ignore-unfixed --ignorefile .trivyignore --skip-files .env /home/docker
```

**Comment by mrexox** [comment]
Right, this should be working in **1.3.10**

You can use `>` to join lines split for readability:

```yml
commands:
  trivy:
    tags: security
    run: >
      docker run --rm --volume "$(git rev-parse --show-toplevel)":/home/docker cdci 
      trivy filesystem --quiet --ignore-unfixed --ignorefile .trivyignore --skip-files .env /home/docker
```

**Comment by iafelix** [comment]
> From a casual reading of your example, I'd expect it to treat separate lines as different commands, since that is how multiple lines in a shell would execute. Using a `\` to span commands multiple lines might work:
> 

That's the case sometimes more than one command is required to perform the validation. 

For ansible-lint for example, I need to run one command to store the credentials for private repository, then run the ansible-galaxy to install assets from that repo, and finally run the ansible-lint. It's expected that any non-zero return code from multiple commands to fail the validation. 

I could do it in separated runs but then I would loose the capability to run multiple tests in parallel.

**Comment by iafelix** [comment]
> Right, this should be working in **1.3.10**
> 
> You can use `>` to join lines split for readability:

For a single command that works perfectly.

**Comment by mrexox** [comment]
I've tested this for multiple commands, and it works well. Can you test it on your setup?

```yml
# lefthook.yml

test:
  commands:
    fail:
      run: |
        exit 1
        echo never prints
    success:
      run: |
        echo line 1
        echo line 2
```

Output:

```$ lefthook run test
Lefthook v1.3.10
RUNNING HOOK: test

  EXECUTE > fail


  EXECUTE > success
line 1
line 2


SUMMARY: (done in 0.01 seconds)
✔️  success
🥊  fail
```

**Comment by iafelix** [comment]
Hey,

Sorry for late response!
I don't remember on which previous version it didn't work, or there's a chance that I've made something wrong too lol. 

Well it's working perfectly fine. 

Thanks!

**Comment by mrexox** [comment]
@Gonkers , could you please show the logs with running `lefthook run test -v`?
